leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/LwM2mDecoder.java

```diff
@@ -97,6 +98,19 @@
     List<TimestampedLwM2mNode> decodeTimestampedData(byte[] content, ContentFormat format, LwM2mPath path,
             LwM2mModel model) throws CodecException;
 
+    /**
+     * Deserializes a binary content into a {@link TimestampedLwM2mNodes}.
+     * <p>
+     *
+     * @param content the content
+     * @param format the content format
+     * @param model the collection of supported object models
+     * @return the decoded timestamped nodes represented by {@link TimestampedLwM2mNodes}
+     * @throws CodecException if content is malformed.
+     */
+    TimestampedLwM2mNodes decodeTimestampedNodes(byte[] content, ContentFormat format, LwM2mModel model)
+            throws CodecException;
+
```

Comment:
Compared to decodeNodes API:
```java
Map<LwM2mPath, LwM2mNode> decodeNodes(
                   byte[] content,
                   ContentFormat format,
                   List<LwM2mPath> paths,
                   LwM2mModel model)  throws CodecException;
```
We removed `List<LwM2mPath> paths`but I guess this will be necessary again when we will implement  https://github.com/eclipse/leshan/issues/1089 :thinking:

Reply:
What if the ```List<LwM2mPath> paths``` parameter would be implemented as filter of result? It could simplify decoder implementation a little bit.

Reply:
The idea is exaclly the same as decodeNodes.

For observe we need the list of paths for send we don't. 

But maybe we add it later when/if we implement timestamp for composite observe.

Reply:
I looked at builder update and the solution is great for me :+1: